### ISSUE TITLE
Testing perl CORE leaves a trail of junk in /tmp

### ISSUE THREAD

**Issue by Tux** [Test-More/test-more#673]
``` sh
$ find test2*|wc -l
1682
```
lib/Test2/IPC/Driver/Files.pm explicitly sets CLEANUP => 0 due to forking
but none of the processes clears up after itself

Running perl CORE smokes for a couple of days will fill up /tmp

**Comment by exodist** [comment]
reproduction case: dist/Thread-Queue/t/07_lock.t  this test in the perl test suite is one that has leftover files.

**Comment by exodist** [comment]
#674

**Comment by exodist** [comment]
I have identified a test in Test2's suite that leaves 2 dirs in place. This test is easily fied, and I have a solution in #674. Tomorrow I will submit a patch to p5p that fixes this specific test.

I have also identified a test in perl-core that leaves the files behind. This is more tricky, the problem is that it starts a thread, then detaches the thread. The parent thread exits before the child thread. The problem here is that this prevents global destruction from happening. Global destruction is where the cleanup happens.

For now I will submit a patch to p5p that directly fixes this specific test, it appears to be a one-off in the perl test suite, so fixing it directly can at least stop the tmp fill-up in blead.

I have been talking on #p5p looking for a test2-specific fix/workaround to address the detached thread issue, I have a patch, but I do not like it. I am going to take the conservative approach and hold back that patch for now. If detaching a child and letting the parent thread exit is common, and deemed sane then I will put a work around in place, though I am hoping we can find a better one by then.
